### PR TITLE
Fix for eager alter statements w/ tests.

### PR DESCRIPTION
**Pull request by stratease**
[TEC-4797](https://theeventscalendar.atlassian.net/browse/TEC-4797)

Removing our eager schema updates. This was causing a number of `ALTER` statements being run redundantly. No longer utilizes cache/transient for the last run check as it is not dependable.

Artifacts:
```
ALTER TABLE wp_tec_occurrences CHANGE COLUMNstart_date start_date VARCHAR(19) NOT NULL
```

See tests.

[TEC-4797]: https://theeventscalendar.atlassian.net/browse/TEC-4797?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Field.php [hunk 1]
```diff
@@ -26,13 +26,21 @@ abstract class Abstract_Custom_Field implements Field_Schema_Interface {
...
 		$results = $this->after_update( $results );
 
 		return $results;
 	}
 
+	public function validate_for_dbDelta( string $query ) {
+		if ( preg_match( '/`.*?` [A-Z]/', $query ) ) {
+			do_action( 'tribe_log', 'error', __METHOD__, ['schema_builder_error' => "Failed dbDelta field validation: $query"] );
+		}
+	}
```
Comment (bordoni) [inline src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Field.php:42]: Weird naming, can we add docblocks here too?

File: src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php [hunk 2]
```diff
@@ -55,13 +55,21 @@ public function empty_table() {
...
 		$results = $this->after_update( $results );
 
 		return $results;
 	}
 
+	public function validate_for_dbDelta( string $query ) {
+		if ( preg_match( '/`.*?` [A-Z]/', $query ) ) {
+			do_action( 'tribe_log', 'error', __METHOD__, ['schema_builder_error' => "Failed dbDelta field validation: $query"] );
+		}
+	}
```
Comment (bordoni) [inline src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php:71]: Weird naming, can we add docblocks here too?

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php [hunk 3]
```diff
@@ -55,13 +55,28 @@ public function empty_table() {
...
 	}
 
+	/**
+	 * Inspects query strings being passed to dbDelta, and logs an error if not ideal.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $query Query string to inspect for case sensitivity before using in dbDelta
+	 */
+	public function validate_for_dbDelta( string $query ) {
```
Comment (bordoni) [inline src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php:74]: Avoid the camelcase method.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php (hunk 3 above)
Comment (stratease) [reply src/Events/Custom_Tables/V1/Schema_Builder/Abstract_Custom_Table.php:74]: The function it referenced is camel cased.

**Review by bordoni — APPROVED** [review]

**Pull request merged**